pkg/flavors/flavors.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package flavors
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+)
+
+// FlavorConfigTemplate - interface to individual flavors.
+type FlavorConfigTemplate interface {
+	CheckParseParams(params map[string]string) error
```

Comment:
Sorry if I was ambiguous.

I was proposing to rename `CheckParams()` into `Parse<something>()`.

`CheckParseParams()` is too complicated for me :)